src/iguana/algorithms/clas12/PhotonGBTFilter.cc

```diff
@@ -0,0 +1,417 @@
+#include "PhotonGBTFilter.h"
+
+namespace iguana::clas12 {
+
+  REGISTER_IGUANA_ALGORITHM(PhotonGBTFilter);
+
+  void PhotonGBTFilter::Start(hipo::banklist& banks)
+  {
+    ParseYAMLConfig();
+      
+    b_particle = GetBankIndex(banks, "REC::Particle");
+    b_calorimeter = GetBankIndex(banks, "REC::Calorimeter");
+    b_config = GetBankIndex(banks, "RUN::config");
+    
+    o_pass      = GetCachedOption<int>("pass").value_or(0);
+    o_threshold = GetCachedOption<double>("threshold").value_or(0);
+    
+  }
+
+
+    
+  void PhotonGBTFilter::Run(hipo::banklist& banks) const
+  {
+      
+    auto& particleBank = GetBank(banks, b_particle, "REC::Particle");
+    auto& caloBank     = GetBank(banks, b_calorimeter, "REC::Calorimeter");
+    auto& configBank = GetBank(banks,b_config,"RUN::config");
+    int runnum = configBank.getInt("run",0);
+       
+    // Get CaloMap for the event
+    auto calo_map = GetCaloMap(caloBank);
+    
+    // dump the bank
+    ShowBank(particleBank, Logger::Header("INPUT PARTICLES"));
+      
+    // Loop over each photon in the particleBank to classify it
+    for(int row = 0; row < particleBank.getRows(); row++) {
+      auto pid    = particleBank.getInt("pid", row);
+      if (pid!=22) continue;
+      auto accept = Filter(particleBank, caloBank, calo_map, row, runnum);
+      if(!accept)
+        MaskRow(particleBank, row);
+    }
```

Comment:
HIPO now supports bank filtering, so `MaskRow` has been removed:
```suggestion
    particleBank.getMutableRowList().filter([this](auto bank, auto row) {
      auto pid = bank.getInt("pid", row);
      if (pid!=22) return true;
      return Filter(bank, caloBank, calo_map, row, runnum);
    });
```